std-bits/table/src/main/java/org/enso/table/data/column/operation/map/DoubleLongMapOpWithSpecialNumericHandling.java

```diff
@@ -0,0 +1,38 @@
+package org.enso.table.data.column.operation.map;
+
+import java.util.BitSet;
+import org.enso.table.data.column.storage.numeric.DoubleStorage;
+import org.enso.table.data.column.storage.numeric.LongStorage;
+
+public abstract class DoubleLongMapOpWithSpecialNumericHandling
+    extends UnaryMapOperationWithProblemBuilder<Double, DoubleStorage> {
+  public DoubleLongMapOpWithSpecialNumericHandling(String name) {
+    super(name);
+  }
+
+  protected abstract long doOperation(double a);
+
+  @Override
+  public LongStorage run(
+      DoubleStorage storage, Object arg, MapOperationProblemBuilder problemBuilder) {
+    long[] out = new long[storage.size()];
+    BitSet isMissing = new BitSet();
+
+    for (int i = 0; i < storage.size(); i++) {
+      if (!storage.isNa(i)) {
+        double item = storage.getItem(i);
+        boolean special = Double.isNaN(item) || Double.isInfinite(item);
+        if (!special) {
+          out[i] = doOperation(item);
+        } else {
+          String msg = "Value is " + item;
```

Comment:
This error message needs to be more user friendly.

```suggestion
          String msg = "The operation " + name + " does not allow " + item + " as input, returning Nothing.";
```
or sth like that